_studio/shared/src/mfx_umc_mjpeg_vpp.cpp

```diff
@@ -217,8 +217,16 @@ mfxStatus mfx_UMC_FrameAllocator_D3D_Converter::StartPreparingToOutput(mfxFrameS
                 isD3DToSys = !((pOpaqAlloc->Out.Type & MFX_MEMTYPE_SYSTEM_MEMORY) == 0);
             }
 
-            pSrc.Info.CropW = surface_work->Info.CropW;
-            pSrc.Info.CropH = surface_work->Info.CropH;
+            if (par->mfx.Rotation == MFX_ROTATION_0 || par->mfx.Rotation == MFX_ROTATION_180)
+            {
+                pSrc.Info.CropW = surface_work->Info.CropW;
+                pSrc.Info.CropH = surface_work->Info.CropH;
+            }
+            else
+            {
+                pSrc.Info.CropW = surface_work->Info.CropH;
```

Comment:
You assign CropW = CropH. Why you don't do the same for Width = Height?

Reply:
Let me answer. For some cases including rotation case, JPEG decoder works in 2 steps: decoding step and VPP step. This is not common MSDK VPP but internal JPEG decoder VPP and it is not seen for the application. So we need a set of some internal surfaces to share them between decoder and VPP.  pSrc is one of such surfaces. Width and height of pSrc are initialized in decoder Init and are correct. But in some cases the real size of the frame is lower than allocated surface and so we need to set crop values without decoder re-initialization, that is why we copy crop values here (but not width and height).
In case of rotation, surface_work has rotated CropW and CropH, but pSrc should have not rotated crops. That is why they are swapped here. Width and height should not be changed here

Reply:
ok. This explains Width and Height. How about CropX and CropY :)?

Reply:
JPEG standard does not support crops as it is done in AVC, so
- CropX and CropY are always 0,
- CropW and CropH represents picture size for current frame,
- Width and Height represents surface allocation size

Reply:
This would make very good comment in the code and avoid similar questions in the future. @Alexandr-Konovalov : can you, please, add it to the code? After that I will be ok with the change.

Reply:
@dvrogozh Sure.